### PR TITLE
RARO-424: Eissn tag blank in preprint xml and accepted by Rhino 

### PR DESCRIPTION
**Pull request by sjtower**

### CODE REVIEW

File: src/main/java/org/ambraproject/rhino/model/article/ArticleMetadata.java [hunk 1]
```diff
@@ -311,39 +322,42 @@ public boolean equals(Object o) {
...
-    if (publisherLocation != null ? !publisherLocation.equals(that.publisherLocation) : that.publisherLocation != null) {
+    if (publisherLocation != null ? !publisherLocation.equals(that.publisherLocation) : that.publisherLocation != null)
+      return false;
+    if (publisherName != null ? !publisherName.equals(that.publisherName) : that.publisherName != null)
       return false;
-    }
-    if (publisherName != null ? !publisherName.equals(that.publisherName) : that.publisherName != null) return false;
     if (url != null ? !url.equals(that.url) : that.url != null) return false;
-    if (nlmArticleType != null ? !nlmArticleType.equals(that.nlmArticleType) : that.nlmArticleType != null) {
+    if (nlmArticleType != null ? !nlmArticleType.equals(that.nlmArticleType) : that.nlmArticleType != null)
```
Comment (randy-tano) [inline src/main/java/org/ambraproject/rhino/model/article/ArticleMetadata.java]: +1 to one of the most creative use of the **ternary** operator, haven't seen it in an if-statement.  I guess it boils down to:

```
if (nlmArticleType != null && !nlmArticleType.equals(that.nlmArticleType)) {
 return false;
}
```
Not necessary to have **that.nlmArticleType != null**, as **!nlmArticleType.equals(that.nlmArticleType)** will handle that case.

File: src/main/java/org/ambraproject/rhino/config/YamlConfiguration.java [hunk 2]
```diff
@@ -108,6 +112,11 @@ public String getDefaultBucket() {
       public ImmutableSet<String> getAllBuckets() {
         return allBuckets;
       }
+
+      @Override
+      public ImmutableSet<String> getSecondaryBuckets() {
```
Comment (randy-tano) [inline src/main/java/org/ambraproject/rhino/config/YamlConfiguration.java]: Nice on the ImmutableSet!

File: src/main/java/org/ambraproject/rhino/model/article/ArticleMetadata.java (hunk 1 above)
Comment (sjtower) [reply src/main/java/org/ambraproject/rhino/model/article/ArticleMetadata.java]: This code is all auto-generated by IntelliJ so I can't claim any credit :)
Yes, it is unoptimized, but we typically don't mess with the auto-generated equals/hashcode cause we consider it "under the covers" so to speak.

**Comment by sjtower** [comment]
Leaving for @msingh05 to take a look at.

**Comment by sjtower** [comment]
See also: https://github.com/PLOS/developer-scripts/pull/130

**Pull request merged**